benchmarks/NN/oop_c#/Program.cs

```diff
@@ -0,0 +1,400 @@
+
+            int nInputs = train.GetLength(1);
+            int nOutputs = trainActual.Distinct().Count();
+
+            NeuralNetwork network = new NeuralNetwork(new AccuracyPercentage()).InitialiseLayers(nInputs, N_HIDDEN, nOutputs);
+            network.Train(train, trainActual, LEARNING_RATE, N_EPOCHS);
+            (int[] predictions, double accuracy) = network.Predict(test, testActual);
+
+            System.Console.WriteLine(accuracy);
+        }
+    }
+
+    public static class Utils
+    {
+        public static double[,] LoadCSV(string filepath)
+        {
+            var file = File.ReadAllLines(filepath);
+            double[,] dataset = new double[file.Length, file[0].Split(',').Length];
+            for (int i = 0; i < file.Length; i++)
+            {
+                var values = file[i].Split(',');
+                for (int j = 0; j < values.Length; j++)
+                {
+                    if (j == values.Length - 1)
+                        dataset[i, j] = double.Parse(values[j]) - 1;
+                    else
+                        dataset[i, j] = double.Parse(values[j]);
+                }
+            }
+            return dataset;
+        }
+
+        public static double[,] NormaliseColumns(double[,] dataset)
+        {
+            int numColumns = dataset.GetLength(1) - 1;
+            for (int i = 0; i < numColumns; i++)
+            {
+                double[] column = dataset.GetCol(i);
+                double min = column.Min();
+                double max = column.Max();
+                double diff = max - min;
+                for (int j = 0; j < column.Length; j++)
+                    dataset[j, i] = (dataset[j, i] - min) / diff;
+            }
+            return dataset;
+        }
+
+        public static (double[,], double[,]) GetTestTrainSplit(double[,] dataset, double percent)
+        {
+            dataset = shuffle(new Random(), dataset);
+            (int rows, int columns) = (dataset.GetLength(0), dataset.GetLength(1));
+            int numTest = (int)(rows * percent);
+            int numTrain = rows - numTest;
+            double[,] test = new double[numTest, columns];
+            double[,] train = new double[numTrain, columns];
+            int testindex = 0;
+            int trainindex = 0;
+            for (int i = 0; i < rows; i++)
+            {
+                if (i < numTest)
+                    test.SetRow(testindex++, dataset.GetRow(i));
+                else
+                    train.SetRow(trainindex++, dataset.GetRow(i));
+            }
+            return (test, train);
+        }
+
+        private static T[,] shuffle<T>(Random random, T[,] array)
+        {
+            T[,] shuffledArray = new T[array.GetLength(0), array.GetLength(1)];
+            int numRows = array.GetLength(0);
+            int[] indicies = new int[numRows];
+            Random rnd = new Random();

```

Comment:
```suggestion
            Random rnd = new Random(2);

```
Plejer vi ikke at have seed på vores random, så vi er sikre på det er det samme vi udregner i alle benchmarks?

Reply:
På den anden side kan jeg jo også se at jeg sortere på en anden måde uanset...